src/models/message-trait.ts

```diff
@@ -6,7 +6,8 @@ import type { SchemaInterface } from './schema';
 
 export interface MessageTraitInterface extends BaseModel, BindingsMixinInterface, DescriptionMixinInterface, ExtensionsMixinInterface, ExternalDocumentationMixinInterface, TagsMixinInterface {
   id(): string;
-  schemaFormat(): string;
+  hasSchemaFormat(): boolean;
+  schemaFormat(): string | undefined;
```

Comment:
What do you mean by "t does NOT play well together"?

Reply:
Cause the difference between v2 and v3 is that message traits no longer has schemaFormat, at all... And with inheritance we force it. 

But I am not sure whether it would be better with no inheritance anyway, cause the parser API would have to change regardless.

Not fan of the `hasSchemaFormat` and `schemaFormat(): string | undefined;` but I see no other way around it.

Reply:
I think `hasSchemaFormat()` not being present in the Parser-API was a mistake. Same for the `schemaFormat(): string | undefined` instead of just `string`.

Reply:
Will you update the Parser-API then @jonaslagoni ?

Reply:
`schemaFormat(): string` can stay. We should return the default schema format if not present.

Reply:
cc @fmvilas so you can read my last comment here

Reply:
So MessageObject doesn't have `schemaFormat` anymore on v3 but it's on v2. On v3, we get it from the schema the message is using as a payload. In both cases, we have defaults.

However, this is not about the MessageObject but about the MessageTraitObject. So, I'm thinking we probably shouldn't have `format()` and we should have `schemaFormat()` for backward compatibility, which could return the value of the `schemaFormat` field (including undefined). In summary:

```
schemaFormat(): string | undefined // in v3 this is always undefined
hasSchemaFormat(): boolean // in v3 this is always false
```

Reply:
Ok, I think we are talking about the same but It seems I didn't use the right words. Here is the final picture Im talking about:

### MessageTrait 
schemaFormat(): string | undefined
hasSchemaFormat(): boolean

### Message:
schemaFormat(): string

### Schema:
format(): string

Reply:
Actually, `message.schemaFormat()` should also allow return `undefined` since `payload` is not mandatory.

I'm thinking that, for simplification purpose, I would remove it from message object and just leave the one in the schema (`format() : string`), so people can get the format from the `message.payload().format()`.

The final pic will then be:

### MessageTrait 
schemaFormat(): string | undefined
hasSchemaFormat(): boolean

### Schema:
format(): string

WDYT @fmvilas @jonaslagoni @magicmatatjahu ?